x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/compare_synced_integrations.ts

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { isEqual } from 'lodash';
+import type {
+  ElasticsearchClient,
+  Logger,
+  SavedObjectsClientContract,
+  SavedObjectsFindResult,
+} from '@kbn/core/server';
+
+import type {
+  CcrFollowInfoFollowerIndex,
+  ClusterComponentTemplateSummary,
+  IngestGetPipelineResponse,
+} from '@elastic/elasticsearch/lib/api/types';
+
+import { appContextService } from '../../services';
+import { getPackageSavedObjects } from '../../services/epm/packages/get';
+
+import { IndexNotFoundError } from '../../errors';
+
+import type { Installation } from '../../types';
+
+import type {
+  RemoteSyncedIntegrationsStatus,
+  GetRemoteSyncedIntegrationsStatusResponse,
+  SyncStatus,
+  RemoteSyncedCustomAssetsStatus,
+  RemoteSyncedCustomAssetsRecord,
+} from '../../../common/types';
+
+import type { IntegrationsData, SyncIntegrationsData, CustomAssetsData } from './model';
+import { getPipeline, getComponentTemplate, CUSTOM_ASSETS_PREFIX } from './custom_assets';
+import { getFollowerIndex } from './sync_integrations_on_remote';
+
+export const getFollowerIndexInfo = async (
+  esClient: ElasticsearchClient,
+  logger: Logger
+): Promise<{ info?: CcrFollowInfoFollowerIndex; error?: string }> => {
+  try {
+    const index = await getFollowerIndex(esClient, new AbortController());
+    if (!index) {
+      return { error: `Follower index not found` };
+    }
+    const res = await esClient.ccr.followInfo({
+      index,
+    });
+    if (!res?.follower_indices || res.follower_indices.length === 0)
+      return { error: `Follower index ${index} not available` };
+
+    if (res.follower_indices[0]?.status === 'paused') {
+      return { error: `Follower index ${index} paused` };
+    }
+    return { info: res.follower_indices[0] };
+  } catch (err) {
+    if (err?.body?.error?.type === 'index_not_found_exception')
+      throw new IndexNotFoundError(`Index not found`);
```

Comment:
should we return the error message instead of throwing an error?

Reply:
okay, then shouldn't we return the error in the catch there? https://github.com/elastic/kibana/pull/216178/files#diff-f8de6a6d308d65b9d61c400a2fdbebe1078e67eaf036537c028476260be254f5R324

Reply:
Fixed, I added some unit tests to cover these cases